src/commands/CommandsManager.ts

```diff
@@ -317,7 +316,7 @@ export class CommandsManager {
 	 * @private
 	 */
 	private static shouldSendHelpMessage(message: Message, client: Client): boolean {
-		return message.mentions.has(client.user.id) && this.hasChannelPermission(message.channel as GuildChannel)[0];
+		return message.mentions.has(client.user.id) && this.hasChannelPermission(message.channel as unknown as DraftbotChannel)[0];
```

Comment:
c'est quoi ce as unknown au milieu ?

Reply:
ca permet de dire à ts "oui oui, je veux le cast dans ce type, no worries. Ici je peux m'amuser à cast le guildchan en draftbotchan, mais useless car pas d'appel à send

Reply:
du coup ici c'est useless ?

ya pas un moyen de faire ce cast sans avoir à dire à ts oui oui je veux le faire ?

Reply:
nope car techniquement les fonctions send ont pas la même signature, donc ts va gueuler

Reply:
je comprend pas comment ya pas un moyen de faire un héritage et de surcharger la méthode sans avoir à faire un cast as unknown ? genre au lieu de lancer manuellement les interactions et les canaux dans ton type personnalisé, on pourrait créer des classes qui héritent de ces types et surcharger les méthodes nécessaires.

Reply:
problème, c'est que ca, ca marche UNIQUEMENT si les fonctions en question n'ont qu'un seul héritage possible ET que celles-ci ont la même signature. Ici, non seulement les fonctions n'ont pas les mêmes signatures, mais aussi il y a 2 fonctions pour chacune des fonctions en dessous. De plus, l'interaction possède déjà un channel, donc faut override aussi le type du channel, ce qui n'est pas possible avec un héritage classique

Reply:
again même problème : pas la même signature sur les fonctions et attributs

Reply:
Une autre option pourrait être d'utiliser les méthodes et les propriétés comme des interfaces plutôt que des classes, puis de les implémenter dans nos propres classes ?

Reply:
problème cette fois, c'est que ca force à réécrire toutes les fonctions inchangées de discord, ce qui demande une réécriture bien plus importante que ce que j'ai fait la

Reply:
et ca marcherait pas car certaines fonctions sont privées et non récupérables

Reply:
ok let's go for that then, tu disais que cette occurence était useless tho ?

Reply:
ouais car la le seul truc qui diffère entre les deux c'est la fonction send, or dans cette fonction spécifique on utilise pas la fonction send, donc useless de cast, juste faire une conversion classique is okay

Reply:
tu peux fix ?

Reply:
nope :doge: c'est intentionnel que j'ai fait ainsi

Reply:
> nope :doge: c'est intentionnel que j'ai fait ainsi
je croyais que ct useless sur cette occurence ???